### PR TITLE
Updated newchunk_nolongjmp() 

### PR DESCRIPTION
**Pull request by alexdaniellungu**
Updated newchunk_nolongjmp() to consider capability size. Without this change wpa_supplicant triggers a bus error on Morello.

### CODE REVIEW

File: contrib/libpcap/gencode.c [hunk 1]
```diff
@@ -591,8 +591,8 @@ newchunk_nolongjmp(compiler_state_t *cstate, size_t n)
 	size_t size;
 
 #ifndef __NetBSD__
-	/* XXX Round up to nearest long. */
-	n = (n + sizeof(long) - 1) & ~(sizeof(long) - 1);
+	/* XXX Round up to nearest long or capability size. */
+	n = (n + MAX(sizeof(void *), sizeof(long)) - 1) & ~(MAX(sizeof(void *), sizeof(long)) - 1);
```
Comment (jrtc27) [inline contrib/libpcap/gencode.c:595]: In C11+ (which we can assume in CheriBSD) it should be `_Alignof(max_align_t)` not any of this, and for upstream it should be some C89-compatible `offsetof(struct { char dummy; union { long long x; void *p; } u; }, u)` thing (but not inline). I intend to just upstream a proper fix and then pull it back into FreeBSD+CheriBSD.

Comment (jrtc27) [reply contrib/libpcap/gencode.c:595]: https://github.com/the-tcpdump-group/libpcap/pull/1158

**Comment by guyharris** [comment]
> for upstream it should be some C89-compatible

C99-compatible, as we now require some level of C99 support.

**Comment by brooksdavis** [comment]
I've merged Jessica's version of the fix from the libpcap repo

**Pull request closed without merging**